feat(template): Add dropdown menu to the header

ERosendo opened the pull request:
This PR adds a dropdown menu to the header's **Big Cases** button.

The PR introduces the following changes:

- Adds [Flowbite ](https://github.com/themesberg/flowbite) as a NPM dependency.

- Updates the tailwind.config.js file to add Flowbite as a plugin

- Updates the `dev` and the `build` Tailwind commands to include Flowbite’s JavaScript file directly from the node_modules/ folder, so We use it as a static asset. 

- Adds Flowbite’s JS file inside the base template.

- Adds a dropdown menu to the **Big Cases** button in the header.

- Adds the homepage's link to the logo in the header.

- Adds a new template called `post.html` for the various pages that need text.

- Adds a template for the **about** and the **sponsor**'s page.

- Updates the Sponsorship and History links in the footer.

### Additional comments

I think We should add the SVG icons as inline SVG. We moved the inline SVG into their own files in https://github.com/freelawproject/bigcases2/pull/111 and We're using the `img` tag to load the files in the HTML but this approach doesn't allow us to restyle the image, so If We want to use the same icon with another color We need to add another SVG file.

Using inline SVG has the advantage that allow us to use the utility classes from Tailwind to style it, like changing the fill color or applying pseudoclasses (like :focus). We can also use the Django template language to avoid resource-intensive maintenance of these elements.

mlissner reviewed (changes requested):
This looks pretty good. A couple notes:

1. We should do more namespacing. For example, calling a view "about" isn't very specific. That should be about_big_cases or bc_about or big_cases_about. 

2. The same goes for our templates. Let's create a directory for `simple-pages` and put about.html and sponsors.html in there. But probably those files should get namespaced too: bc_about.html? big_cases_about.html? big-cases/about.html? 

3. As for style, the drop down looks pretty OK, but a few things would make it better:
    - More whitespace around the two links (about and sponsors). Check out how the free.law one is roomier.
    - Less vertical whitespace around the buttons at the bottom
    - The menu should fade in/out. I think that's being done with CSS in free.law. This should help with it looking invisible, but if it doesn't, then it should get a drop shadow or a light border around the outside or both.

mlissner commented:
For the fade-in effect on the menu, we use the headless UI Transition element:

https://headlessui.com/react/transition

I think it just applies tailwind CSS classes though. A quick review led me here: 

https://tailwindcss.com/docs/translate

ERosendo commented on bc/assets/templates/includes/header.html:
```diff
@@ -19,10 +21,59 @@
+                              <li>
+                                <a href="{% url 'about' %}" class="flex items-start px-4 py-2 hover:bg-gray-100 dark:hover:bg-gray-600 dark:hover:text-white">
+                                    <div class="h-6 w-6 relative flex-shrink-0">
+                                        <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" class="w-6 h-6">
+                                            <path stroke-linecap="round" stroke-linejoin="round" d="M11.25 11.25l.041-.02a.75.75 0 011.063.852l-.708 2.836a.75.75 0 001.063.853l.041-.021M21 12a9 9 0 11-18 0 9 9 0 0118 0zm-9-3.75h.008v.008H12V8.25z" />
+                                        </svg>
```
Alright!

ERosendo commented on bc/web/views.py:
```diff
@@ -12,6 +12,16 @@ def view_docket(request: HttpRequest, subscription_id: int) -> HttpResponse:
     )
 
 
+def about(request: HttpRequest) -> HttpResponse:
+    """Loads the about page"""
+    return TemplateResponse(request, "about.html")
+
+
+def sponsor(request: HttpRequest) -> HttpResponse:
+    """Loads the sponsors page"""
+    return TemplateResponse(request, "sponsors.html")
```
Thanks for your suggestion. I'll create the generic views directly in your URLconf.

ERosendo commented:
Thanks for your suggestion. A couple of thoughts:

 1. I think We can use the `templates` folder inside the `Web` app to store the pages of the website (We have the tailwind config in this app) instead of creating a the simple-pages folder.

2. You're right about namespacing. I added a folder called `big-cases` and move the `about.html `and `sponsors.html` files into it.

3. I've used some styles from free.law to update the dropdown and also added a CSS animation to fade in/out the menu.

4. Flowbite was only opening/closing the dropdown menu. I removed it from the project and added a new .js file called `dropdown.js`. This new file contains the functions and listener to show/hide the menu and is 4kb (We can use [jscompress](https://jscompress.com/) to minify it and reduce its size down to 2kb).

let me know what you think.

mlissner commented:
Cool. I think the dropdown.js seems way too complicated. I was expecting maybe 10 lines of code for this, but it works and is good enough for me. Thanks for the extra work on this one.
